src/shell/interpreter.zig

```diff
@@ -10101,6 +10114,734 @@ pub const Interpreter = struct {
                 }
             }
         };
+
+        pub const Cp = struct {
+            bltn: *Builtin,
+            opts: Opts = .{},
+            state: union(enum) {
+                idle,
+                exec: struct {
+                    target_path: [:0]const u8,
+                    paths_to_copy: []const [*:0]const u8,
+                    started: bool = false,
+                    tasks_count: i32 = 0,
```

Comment:
is this thread safe? does it need to be thread safe? why is it an i32 instead of a u32?

Reply:
its thread safe as it is only read and mutated by the main thread, changed it to a u32